File: client/common/bet.go
package common

import (
	"fmt"
	"bytes"
)

type Bet struct {
	agency string
	first_name string
	last_name string
	document string
	birth_date string
	number string
}

func NewBet(agency string, first_name string, last_name string, document string, birth_date string, number string) *Bet {
	bet := &Bet{
		agency: agency,
		first_name: first_name,
		last_name: last_name,
		document: document,
		birth_date: birth_date,
		number: number,
	}
	return bet
}

func (b *Bet) serialize() []byte {
	return []byte(fmt.Sprintf("%s|%s|%s|%s|%s|%s",
	b.agency, b.first_name, b.last_name, 
	b.document, b.birth_date, b.number))
}

func serializeMultipleBets(bets []*Bet) []byte {
	serialized_bets := make([]byte, len(bets))

	for i, bet := range bets {
		if bet != nil {
			serialized_bets[i] = bet.serialize()
		}
	}

	return bytes.Join(serialized_bets, []byte(";"))
}

File: server/common/bet.py
import datetime
BET_MSG_SEPARATOR = "|"
BET_CHUNK_SEPARATOR = ";"

class Bet:
    def __init__(self, bet_agency: str, name: str, last_name: str, document: str, birthdate: str, number: str):
        """
        bet agency must be passed as int
        birthdate fmt: YYYY-MM-DD
        number must be passed ad int
        """
        self.agency = int(bet_agency)
        self.first_name = name
        self.last_name = last_name
        self.document = document
        self.birthdate = datetime.date.fromisoformat(birthdate)
        self.number = int(number)

    @staticmethod
    def deserialize(msg: str):
        """
        Deserialize a message into a Bet object
        """
        fields = msg.split(BET_MSG_SEPARATOR)
        if len(fields) != 6:
            return None
        bet_agency, name, last_name, document, birthdate, number = fields
       
        return Bet(bet_agency, name, last_name, document, birthdate, number)
        
    @staticmethod
    def deserialize_multiple_bets(message: str):
        bets = []
        for bet in message.split(BET_CHUNK_SEPARATOR):
            if not bet:
                continue
            deserialized_bet = Bet.deserialize(bet)
            if deserialized_bet:
                bets.append(deserialized_bet)
        return bets
